src/main/cpp/src/SparkResourceAdaptorJni.cpp

```diff
@@ -1360,6 +1375,11 @@ class spark_resource_adaptor final : public rmm::mr::device_memory_resource {
           }
           transition(thread->second, thread_state::THREAD_RUNNING);
           thread->second.is_cpu_alloc = false;
+          // num_bytes is likely not padded, which could cause slight inaccuracies
+          // but for now it shouldn't matter for watermark purposes
```

Comment:
Please make sure that we document this in the metrics page when it is fully ready

Reply:
Which page are you referring to exactly?

Reply:
Sorry it is an external page https://docs.nvidia.com/spark-rapids/user-guide/latest/tuning-guide.html#metrics